torch/distributed/fsdp/_common_utils.py

```diff
@@ -68,7 +68,10 @@ def _module_handles(state: _FSDPState, module: nn.Module) -> List:
     module is FullyShardedDataParallel, the module._handles will be returned.
     """
     if _is_composable(state):
-        return state._module_to_handles[module][:]
+        assert (
+            module in state._comm_module_to_handles
+        ), f"Expects a `comm_module` but got {module} on rank {state.rank}"
+        return state._comm_module_to_handles[module][:]
```

Comment:
Thanks for fixing this. optimizer state_dict will use this API too.

Reply:
Will module wrapper FSDP also switch to `_comm_modules` in the future?

Reply:
> Will module wrapper FSDP also switch to `_comm_modules` in the future?

I think there is no need for wrapper FSDP. In the current design, each `_comm_module` is exactly the `FullyShardedDataParallel._fsdp_wrapped_module`, which is basically the same thing as the `FullyShardedDataParallel` instance itself, so `module._handles[:]` should be correct.